server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -475,9 +461,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.startObject();
             builder.field("hidden", hidden);
             builder.field(ALLOW_CUSTOM_ROUTING.getPreferredName(), allowCustomRouting);
-            if (DataStream.isFailureStoreFeatureFlagEnabled()) {
-                builder.field(FAILURE_STORE.getPreferredName(), failureStore);
-            }
```

Comment:
Removing this from here creates issues bwc tests qith a previous snapshot version. 

I am added this here so I can add this comment and I will apply the removal on a follow up PR after this gets backported.

My idea is this will stay in 8.x for bwc tests to work but we will stop persisting from now on. This way in `9.x` we can remove it completely as stated here.